Code/ElastixTransformixWrappers/include/sitkElastixImageFilter.h

```diff
@@ -0,0 +1,199 @@
+  std::vector< std::map< std::string, std::vector< std::string > > > GetParameterMap();
+  std::map< std::string, std::vector< std::string > > GetDefaultParameterMap( const std::string transformName, const unsigned int numberOfResolutions = 4, const double finalGridSpacingInPhysicalUnits = 10.0 );
+  unsigned int GetNumberOfParameterMaps();
+
+  SITK_RETURN_SELF_TYPE_HEADER SetParameter( const std::string key, const std::string value );
+  SITK_RETURN_SELF_TYPE_HEADER SetParameter( const std::string key, const std::vector< std::string > value );
+  SITK_RETURN_SELF_TYPE_HEADER SetParameter( const unsigned int index, const std::string key, const std::string value );
+  SITK_RETURN_SELF_TYPE_HEADER SetParameter( const unsigned int index, const std::string key, const std::vector< std::string > value );
+  SITK_RETURN_SELF_TYPE_HEADER AddParameter( const std::string key, const std::string value );
+  SITK_RETURN_SELF_TYPE_HEADER AddParameter( const unsigned int index, const std::string key, const std::string value );
+  SITK_RETURN_SELF_TYPE_HEADER AddParameter( const std::string key, const std::vector< std::string > value );
+  SITK_RETURN_SELF_TYPE_HEADER AddParameter( const unsigned int index, const std::string key, const std::vector< std::string > value );
+  std::vector< std::string > GetParameter( const std::string key );
+  std::vector< std::string > GetParameter( const unsigned int index, const std::string key );
+  SITK_RETURN_SELF_TYPE_HEADER RemoveParameter( const std::string key );
+  SITK_RETURN_SELF_TYPE_HEADER RemoveParameter( const unsigned int index, const std::string key );
+
+  SITK_RETURN_SELF_TYPE_HEADER SetInitialTransformParameterFileName( const std::string initialTransformParmaterFileName );
+  std::string GetInitialTransformParameterFileName();
+  SITK_RETURN_SELF_TYPE_HEADER RemoveInitialTransformParameterFileName();
+
+  std::map< std::string, std::vector< std::string > > ReadParameterFile( const std::string filename );
+  SITK_RETURN_SELF_TYPE_HEADER WriteParameterFile( const std::map< std::string, std::vector< std::string > > parameterMap, const std::string filename );
+
+  Image Execute();
+  std::vector< std::map< std::string, std::vector< std::string > > > GetTransformParameterMap();
+  std::map< std::string, std::vector< std::string > > GetTransformParameterMap( const unsigned int index );
+  Image GetResultImage();
+
+  SITK_RETURN_SELF_TYPE_HEADER PrintParameterMap();
+  SITK_RETURN_SELF_TYPE_HEADER PrintParameterMap( const ParameterMapType parameterMapVector );
+  SITK_RETURN_SELF_TYPE_HEADER PrintParameterMap( const ParameterMapVectorType parameterMapVector );
+
+private:
+
+  class ElastixImageFilterImpl;
+  ElastixImageFilterImpl* m_Pimple;
+
+};
+
+// Procedural Interface
+SITKElastix_EXPORT std::map< std::string, std::vector< std::string > > GetDefaultParameterMap( const std::string transform, const unsigned int numberOfResolutions = 4, const double finalGridSpacingInPhysicalUnits = 8.0 );
```

Comment:
SWIG is generating the following error:
```[Code/ElastixTransformixWrappers/include/sitkElastixImageFilter.h:182](https://itk.org/gitweb?p=SimpleITK.git/blob/master/Code/ElastixTransformixWrappers/include/sitkElastixImageFilter.h#L182): Error: Syntax error - possibly a missing semicolon.```

Look at SimpleITK_Common.i line 162. Similar defines are needed for SITKElastix_EXPORT

Reply:
Thanks, good catch! Should be addressed by the latest [force-push](https://github.com/SimpleITK/SimpleITK/compare/c828029b4d69cc2968d8cc8ae94aa25ec1b7dc86..08a16c9298303c53252d2f6aa40e3384ac4c234b), please check.